### DISCUSSION TITLE
How can I know whether the personalized GEM pass the metabolic tasks

### DISCUSSION THREAD

**Discussion by Joyjoyjoyc** [SysBioChalmers/RAVEN#531] [Help me!] [answered]
Hi,

I’m new here and actually there are some questions confused me quite a long time.
As I followed the instruction of Human-GEM, I’ve combined my own rna-seq with the generic Human-GEM to successfully construct the personalized GEM, but I wonder how to check whether the personalized GEM passed the 57 essential metabolic tasks to make sure its functionality? Besides, I wonder the function ftINIT(), what does the parameter “addedRxnsForTasks” mean?  can that reflect the question as I mentioned before?

Looking forward to your reply!!!

**Comment by JonathanRob** [comment, marked as answer]
Hi @Joyjoyjoyc, if you included the list of 57 tasks as input (to ftINIT) while constructing the personalized GEM, then it should by requirement be able to pass those tasks.

However, if you would like to verify that this is indeed the case, you can use the [`checkTasks`](https://github.com/SysBioChalmers/RAVEN/blob/main/core/checkTasks.m) function - just provide it with your personalized GEM and the task list, and it will report for each task whether it passes or fails.

**Reply by Joyjoyjoyc** [reply]
Thank you so much for your reply!!! maybe there were something wrong with my understanding of the 'metabolic tasks' before. cause in a paper, the author abandoned those personalized GEM which cannot perform all of the metabolic tasks, so i thought maybe it's not a requirement (for ftINT) to pass those tasks, and every GEM should be checked later whether it passed or not. So i still want to know does those abandoned personalized GEM mean, with this sample's rna-seq, there is no reaction involved in 57 tasks, and that's why they failed？

**Reply by JonathanRob** [reply]
There are many approaches to generating personalized GEMs, and how to incorporate metabolic tasks into the process; I'm not sure which paper you're referring to, so I cannot comment on their rationale.

However, it does sound a bit inefficient to generate the personalized GEMs and then abandon/discard those that cannot perform tasks, when one could just implement the tasks as a requirement during the personalization stage, thus not wasting computation time generating models that will anyway be thrown out.

A more logical scenario to me would be if you personalize the GEMs based on some list of essential tasks, but then one can analyze or evaluate (but not necessarily discard) the GEMs' performance on a different task list to get an understanding of their metabolic functionality. Using the same task list for the post-analysis as was used for the initial personalization step would be uninteresting, because all GEMs would be expected to yield identical results (i.e., passing all required tasks).

**Reply by Joyjoyjoyc** [reply]
got it! really appreciate for your reply.

**Reply by JonathanRob** [reply]
Of course - glad I could help!
